src/bridge/mod.rs

```diff
@@ -52,6 +52,17 @@ pub struct Bridge<IS: AsyncRead + AsyncWrite + 'static> {
     joining_map: BTreeMap<String, String>,
 }
 
+fn handle_privmsg_action(text: &String) -> (String, String) {
```

Comment:
```suggestion
fn handle_privmsg_action(text: String) -> (&'static str, String) {
```

Reply:
> Now it's `&String` (not `String`) and still copied. Maybe something about what I was imagining doesn't actually work, or I wasn't clear enough. I've added change suggestions for what I think should work, let me know if it doesn't.

Thanks, I just didn't consider `&'static str` when you mentioned no-copying. :-) Hopefully it now addresses your concern.